specifyweb/frontend/js_src/lib/components/QueryComboBox/index.tsx

```diff
@@ -534,24 +551,29 @@ export function QueryComboBox({
       )}
       {typeof formatted?.resource === 'object' &&
       state.type === 'ViewResourceState' ? (
-        <ResourceView
-          dialog="nonModal"
-          isDependent={field.isDependent()}
-          isSubForm={false}
-          resource={formatted.resource}
-          onAdd={undefined}
-          onClose={(): void => setState({ type: 'MainState' })}
-          onDeleted={(): void => {
-            resource?.set(field.name, null as never);
-            setState({ type: 'MainState' });
-          }}
-          onSaved={undefined}
-          onSaving={
-            field.isDependent()
-              ? f.never
-              : (): void => setState({ type: 'MainState' })
-          }
-        />
+        <ReadOnlyContext.Provider value={state.isReadOnly}>
+          <ResourceView
+            dialog="nonModal"
+            isDependent={field.isDependent()}
+            isSubForm={false}
+            resource={formatted.resource}
+            onAdd={undefined}
+            onClose={(): void => {
+              setState({ type: 'ViewResourceState', isReadOnly: false });
```

Comment:
this is not right. what's the point of calling setState twice?
the 2nd call would just override the 1st call!
and the resource.set... line that was here before is still needed - without it the parent resource would still be pointing to a child resource, which no longer exists

Reply:
ups was looking at a wrong line. the 2nd part of my comment no longer applies (I crossed it out). the first part is still the case and must be fixed.
> this is not right. what's the point of calling setState twice?
the 2nd call would just override the 1st call!